pyramid/config/__init__.py

```diff
@@ -903,6 +921,13 @@ def end(self):
         """
         return self.manager.pop()
 
+    def __enter__(self):
+        self.begin()
+        return self
+
+    def __exit__(self, exc_type, exc_value, exc_traceback):
+        self.end()
```

Comment:
I decided that tracking and avoiding nested commits wasn't our problem and was more trouble than it's worth.

The only other issue was clearing the action state which we do not do on normal commit failures nor do we have an api to do it in general and as such I won't be adding it to this type of commit either until we decide to support it in a standardized way across all commit mechanisms.